crates/rpc/rpc/src/eth/api/server.rs

```diff
@@ -226,6 +226,32 @@ where
             .await?)
     }
 
+    /// Handler for: `eth_getHeaderByNumber`
+    async fn header_by_number(&self, block_number: BlockNumberOrTag) -> Result<Option<Header>> {
+        trace!(target: "rpc::eth", ?block_number, "Serving eth_getHeaderByNumber");
+        let block = EthApi::rpc_block(self, block_number, false).await?;
+
+        let header = match block {
+            Some(block) => block.inner.header,
+            None => return Ok(None),
+        };
+
+        Ok(Some(header))
```

Comment:
this can be solved with:

```rust
        let header = EthApi::rpc_block(self, block_number, false).await?.map(|block|block.inner.header);
        Ok(header)
```